test/blackbox/common/BlackboxTestsPubSubBasic.cpp

```diff
@@ -640,20 +640,19 @@ TEST_P(PubSubBasic, PubSubIncompatibleQosListeners)
     EXPECT_EQ(incompatible_durability_reader.last_incompatible_qos(), eprosima::fastdds::dds::DURABILITY_QOS_POLICY_ID);
     EXPECT_FALSE(incompatible_durability_reader.is_matched());
 
-    // A compatible Reader on the same Topic
-    // Should match and never call incompatible QoS callbacks
+    // A compatible Reader on another Topic
+    // Should not match but never call incompatible QoS callbacks
     // Total count of incompatible QoS occurrences and latest incompatible QoS ID stay the same
-    PubSubReader<HelloWorldType> compatible_reader(TEST_TOPIC_NAME);
+    PubSubReader<HelloWorldType> compatible_reader("IncompatibleTopic");
```

Comment:
Use a define or constexpr for the incompatible topic name, and use it here and below.